src/openklant/management/commands/migrate_to_v2_phonenumbers.py

```diff
@@ -0,0 +1,226 @@
+from django.core.management.base import BaseCommand, CommandParser
+
+import structlog
+from rest_framework.fields import URLValidator
+from rest_framework.reverse import reverse_lazy
+
+from openklant.components.token.models import TokenAuth
+from openklant.migration.client import Client
+from openklant.migration.utils import generate_jwt_token
+from openklant.migration.v1.client import LegacyOpenKlantClient
+from openklant.migration.v1.data import Klant
+from openklant.migration.v2.client import OpenKlantClient
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+KLANT_FIELDS = {field.name: field for field in dataclass_fields(Klant)}
+
+DIGITALE_ADDRESSEN_PATH = reverse_lazy(
+    "klantinteracties:digitaaladres-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+PARTIJEN_PATH = reverse_lazy(
+    "klantinteracties:partij-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+MIGRATION_TOKEN_IDENTIFIER = "Migration application"
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
+
+    if not isinstance(response_data, dict):
+        logger.error("unexpected_response_data", response_data=response_data)
+        return [], None
+
+    items = response_data.get("results", [])
+    klanten = []
+
+    generic_client = Client()
+
+    for data in items:
+        klant = Klant(
+            **{field: value for field, value in data.items() if field in KLANT_FIELDS}
+        )
+
+        if klant.subject and not klant.subject_type:
+            subject_data = generic_client.retrieve(klant.subject)
+
+            if not isinstance(subject_data, dict):
+                logger.error(
+                    "unexpected_response_data",
+                    context="subject_retrieval",
+                    subject_data=subject_data,
+                )
+
+                continue
+
+            klant.set_from_external_subject(subject_data)
+
+        klanten.append(klant)
+
+    next_url = response_data.get("next")
+    return klanten, next_url
+
+
+def _save_klanten_phonenumbers(
+    url: str, token: TokenAuth, klanten: list[Klant]
+) -> list[str]:
+    if not klanten:
+        logger.info("no_klanten_to_save")
+        return []
+
+    logger.info("creating_klanten_v2_api", count=len(klanten))
+
+    openklant_client = OpenKlantClient(url, token)
+    updated_klanten = []
+
+    for klant in klanten:
+        telefoonnummer = klant.to_digitaal_adres_phonenumber()
+
+        if not telefoonnummer:
+            continue
+
+        existing_partijen = openklant_client.retrieve(
+            PARTIJEN_PATH, params={"nummer": klant._get_subject().nummer}
```

Comment:
In order to check the information for the related DigitaleAdressen, you'll have to add `expand=digitaleAdressen` to the params as well